### ISSUE TITLE
a general question about the decoder used in PPLiteSeg (not a bug!)

### ISSUE THREAD

**Issue by mylifeasazucchini** [PaddlePaddle/PaddleSeg#2387]
Hi! 

I wanted to ask about the arm_out_chs parameter used here: https://github.com/PaddlePaddle/PaddleSeg/blob/19eb360a5a7aaa9277345791dfcad34004e84031/paddleseg/models/pp_liteseg.py#L44

From the paper I understand that the authors used the following 2 configs for decoder depending on backbone: STDCNet1- [32, 64, 128]  STDCNet2- [64, 96, 128]

but could you please explain why these particular channel numbers are used ? are they somehow dependent on the number of feature channels coming from the backbone STDCNet1/2 ?

**Comment by mylifeasazucchini** [comment]
Update: Figured that out! But new question about the final convolution and upscaling in the segmentation head of PPLiteSeg: https://github.com/PaddlePaddle/PaddleSeg/blob/19eb360a5a7aaa9277345791dfcad34004e84031/paddleseg/models/pp_liteseg.py#L123

the **inputs** to the final segmentation head is as such (which helped me answer my earlier question):
_torch.Size([16, 32, 64, 64])
torch.Size([16, 64, 32, 32])
torch.Size([16, 128, 16, 16])_

the segmentation head performs 3x3 conv to some intermediate no. of channels followed by a 1x1 conv to reduce number of classes to 4 (4=number of classes that I defined here) thereby,

the **outputs** of this operation produces a list of logits of the following shape:
_torch.Size([16, 4, 512, 512])
torch.Size([16, 4, 512, 512])
torch.Size([16, 4, 512, 512])_

but I was expecting my segmentation head output to be a one element list with shape **16x4x512x512** ? keeping in mind that my Ground Truth label is just one long tensor of **16x1x512x512** ? or did I misunderstanding something here entirely ? :/ 

And if we are only gonna use the 1st element of this list, then couldn't we save on compute on the final segmentation head by just conv + upscaling the 32 channel tensor 

Any inputs will be cool! Thanks!!!

**Comment by juncaipeng** [comment]
Please refer to: https://github.com/PaddlePaddle/PaddleSeg/blob/release/2.6/paddleseg/models/pp_liteseg.py#L116

In training stage, the model has three loss (two auxiliary loss), so there are three outputs. In inference stage, the model only output the final logits.

**Comment by mylifeasazucchini** [comment]
That is exactly the detail that I had missed all along as it was not so clear on the paper! Thank you so much for clearing it up and answering super quick!
